### PR TITLE
Adding notebook to demonstrate crossencoder reranking

### PR DESCRIPTION
**Pull request by thomasht86**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

There have been some Stack Overflow and Slack questions related to this. 
The main goal is to have a pyvespa example for global reranking with crossencoders that people can use as a starting point.
It also was very useful for my own learning. 

Please go through and suggest improvements. 

- I linked to a separate Colab-notebook, do you think this is OK, or should I bring the code into the notebook? 
As it is more like an optional link for the interested, it would be nice to not introduce the extra dependencies, on the other hand, the colab should maybe be under source control as well...?

### CODE REVIEW

File: docs/sphinx/source/examples/cross-encoders-for-global-reranking.ipynb [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+      "{\"id\":\"id:doc:doc::1\",\"fields\":{\"text\":\"The novel 'Moby-Dick' was written by Herman Melville and first published in 1851. It is considered a masterpiece of American literature and deals with complex themes of obsession, revenge, and the conflict between good and evil.\"}}\n",
+      "{\"id\":\"id:doc:doc::5\",\"fields\":{\"text\":\"'The Great Gatsby', a novel written by American author F. Scott Fitzgerald, was published in 1925. The story is set in the Jazz Age and follows the life of millionaire Jay Gatsby and his pursuit of Daisy Buchanan.\"}}\n",
+      "{\"id\":\"id:doc:doc::3\",\"fields\":{\"text\":\"Jane Austen was an English novelist known primarily for her six major novels, which interpret, critique and comment upon the British landed gentry at the end of the 18th century.\"}}\n",
+      "{\"id\":\"id:doc:doc::0\",\"fields\":{\"text\":\"'To Kill a Mockingbird' is a novel by Harper Lee published in 1960. It was immediately successful, winning the Pulitzer Prize, and has become a classic of modern American literature.\"}}\n"
+     ]
+    }
+   ],
+   "source": [
+    "!vespa visit"
```
Comment (kkraune) [inline docs/sphinx/source/examples/cross-encoders-for-global-reranking.ipynb:592]: Need to install the cli wrapper first right?

**Review by kkraune — COMMENTED** [review]
I ran the notebook and the relevance was not that great (I expected the best hits with highest relevance), so it did not illustrate the point, I think - we should look into

**Comment by thomasht86** [comment]
> I ran the notebook and the relevance was not that great (I expected the best hits with highest relevance)

Didn't you get this result ?

```python
("'To Kill a Mockingbird' is a novel by Harper Lee published in 1960. It was "
 'immediately successful, winning the Pulitzer Prize, and has become a classic '
 'of modern American literature.')
0.9634037778787636
("Harper Lee, an American novelist widely known for her novel 'To Kill a "
 "Mockingbird', was born in 1926 in Monroeville, Alabama. She received the "
 'Pulitzer Prize for Fiction in 1961.')
0.8672221280618897
("'The Great Gatsby', a novel written by American author F. Scott Fitzgerald, "
 'was published in 1925. The story is set in the Jazz Age and follows the life '
 'of millionaire Jay Gatsby and his pursuit of Daisy Buchanan.')
0.09325768904619067
("The novel 'Moby-Dick' was written by Herman Melville and first published in "
 '1851. It is considered a masterpiece of American literature and deals with '
 'complex themes of obsession, revenge, and the conflict between good and '
 'evil.')
0.010269765303083865
```

File: docs/sphinx/source/examples/cross-encoders-for-global-reranking.ipynb [hunk 2]
```diff
@@ -0,0 +1,698 @@
...
+   "source": [
+    "## Inspecting the model\n",
+    "\n",
+    "It is useful to inspect the expected inputs and outputs, along with their shapes, before integrating the model into Vespa.\n",
+    "\n",
+    "This can either be done by for instance by using the `sentence_transformers` and `onnxruntime` libraries.\n",
+    "\n",
+    "One-off tasks like this are well suited for a Colab notebook, one example on how to do this in colab can be found here:\n",
+    "\n",
+    "[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/drive/1DfubYQNyBWzBpgyGUBPN5gaIuSJZ28wy)\n",
```
[the remark this answers is not in the document]
Comment (thomasht86) [reply docs/sphinx/source/examples/cross-encoders-for-global-reranking.ipynb:123]: Added header, some intro text and link to the pyvespa example notebook html.

**Review by kkraune — APPROVED** [review]

**Comment by jobergum** [comment]
This is fantastic! Should be a blog post too. I would also like to see a cloud version :) 

I would maybe drop the section onnx analyze tool, since it uses a tool which is used for backend evaluation, with global phase, it's actually eveluated in the stateless container. I would also mention the very important aspect of scaling with sequence length so having d[512] is 4x as expensive as d[128]

**Review by jobergum — APPROVED** [review]

**Comment by kkraune** [comment]
Great feedback, Jo Kr - lets add in subsequent PRs, there are more PRs coming in for other notebooks, so will merge this now

**Pull request merged**